src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,79 @@
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final String INITIALIZE_ERROR = "Can`t initialize field value. Class: ";
+    private static final String INITIALIZE_ERROR_FIELD = ". Field: ";
+    private static final String CREATE_NEW_INSTANCE_ERROR_CLASS = "Class ";
+    private static final String CREATE_NEW_INSTANCE_ERROR_ANNOTATION =
+            " don`t has annotation @Component.";
+    private static final String CREATE_NEW_INSTANCE_ERROR = "Can`t create new instance of ";
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        Object clazzImplementationInstance = null;
+        for (Field field: declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(INITIALIZE_ERROR + clazz.getName()
+                            + INITIALIZE_ERROR_FIELD + field.getName());
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(CREATE_NEW_INSTANCE_ERROR_CLASS + clazz.getName()
+                    + CREATE_NEW_INSTANCE_ERROR_ANNOTATION);
+        }
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(CREATE_NEW_INSTANCE_ERROR + clazz.getName());
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementation = new HashMap<>();
+        interfaceImplementation.put(ProductService.class, ProductServiceImpl.class);
+        interfaceImplementation.put(FileReaderService.class, FileReaderServiceImpl.class);
```

Comment:
you can make this map a class-level variable

also you can use Map.of(key, value, key, value) method - it will create a immutable collection